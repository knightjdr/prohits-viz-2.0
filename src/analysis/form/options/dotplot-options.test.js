import React from 'react';
import { shallow } from 'enzyme';

import DotplotOptions from './dotplot-options';

const form = {
  analysisType: 'test',
  conditionClustering: 'conditions',
  clustering: 'test',
  control: 'test',
  ctrlSub: true,
  normalization: 'none',
  readoutClustering: 'readouts',
  readoutLength: 'test',
  readoutLengthNorm: false,
  score: 'test',
  scoreType: 'lte',
};

describe('DotplotOptions', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <DotplotOptions
        change={jest.fn()}
        form={form}
        header={['a', 'b']}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
