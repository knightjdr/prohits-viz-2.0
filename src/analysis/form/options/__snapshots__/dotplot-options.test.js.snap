// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotplotOptions should match snapshot 1`] = `
<div>
  <ScoreTypeContainer
    analysisType="test"
    change={[Function]}
    score="test"
    scoreType="lte"
  />
  <PrimaryFilterContainer
    analysisType="test"
    change={[Function]}
    fileType="saint"
    score="test"
  />
  <SecondaryFilterContainer
    analysisType="test"
    change={[Function]}
    fileType="saint"
    score="test"
  />
  <MinimumAbundance
    analysisType="test"
  />
  <MaximumAbundance
    analysisType="test"
  />
  <ControlSubtractionContainer
    analysisType="test"
    change={[Function]}
    control="test"
    ctrlSub={true}
    header={
      Array [
        "a",
        "b",
      ]
    }
  />
  <ReadoutLengthNormalizationContainer
    analysisType="test"
    change={[Function]}
    header={
      Array [
        "a",
        "b",
      ]
    }
    readoutLength="test"
    readoutLengthNorm={false}
  />
  <Normalization
    analysisType="test"
    normalization="none"
  />
  <Log
    analysisType="test"
  />
  <FillColor
    analysisType="test"
  />
  <EdgeColor
    analysisType="test"
  />
  <Clustering
    analysisType="test"
    clustering="test"
    conditionClustering="conditions"
    readoutClustering="readouts"
  />
  <Output
    analysisType="test"
  />
</div>
`;
